app/models/enrollment/recent_activity.rb

```diff
@@ -52,9 +52,12 @@ def total_activity_interval(as_of)
     end
 
     def update_with(options)
-      result = all_enrollments_scope.update_all(options)
-      
-      all_enrollments_scope.ids.each do |id|
+      active_scope = all_enrollments_scope
```

Comment:
How does filtering out deleted stop a dead lock? Since we would be filtering out separate records anyways.